src/features/home/carousels/platforms/components/item_hero.image.tsx

```diff
@@ -0,0 +1,76 @@
+import { graphql, useStaticQuery } from 'gatsby'
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+import { TPlatformImageKeys, TPlatformsDetailImagesQuery } from '../types'
+import { QueryImage } from 'components/elements'
+import { useCountryRule } from 'components/hooks/use-country-rule'
+
+const MobileImage = styled(QueryImage)`
+    .gatsby-image-wrapper > picture > img {
+        object-fit: contain !important;
+    }
+`
+
+type ItemHeroImageProps = {
+    image_key: TPlatformImageKeys
+}
+
+const ItemHeroImage = ({ image_key }: ItemHeroImageProps) => {
+    const images_data = useStaticQuery<TPlatformsDetailImagesQuery>(query)
+    const rules = useCountryRule()
+
+    const platform_image_key = useMemo(() => {
+        return typeof image_key === 'function' ? image_key(rules) : image_key
```

Comment:
there will be several hooks later on which they'll get a property of this object and then return the appropriate value for the property. in the future this might change to the `image_key_extractor`, but for now I don't know the full requirements of those hooks.
those hooks will be generic and they'll have kinda the same signature of `useVisibleContent` hook, but for one property. so that's why I don't wanna have 2 points of truth for one property. we can talk more on this if you want :pray: